lib/sidekiq.rb

```diff
@@ -150,7 +150,7 @@ def self.default_server_middleware
     Middleware::Chain.new do |m|
       m.add Middleware::Server::Logging
       m.add Middleware::Server::RetryJobs
-      if defined?(::ActiveRecord::Base)
+      if defined?(::ActiveRecord::Base) && (!defined?(Sidekiq::Rails::Reloader) || Sidekiq.options[:reloader].is_a?(Sidekiq::Rails::Reloader))
```

Comment:
I've combined the conditional to match the one in the middleware and negated the whole thing.